modules/codegen/src/smithy4s/codegen/internals/LineSegment.scala

```diff
@@ -73,6 +73,13 @@ private[codegen] object LineSegment {
   }
 
   object NameRef {
+    val autoImportedNames: List[String] = List(
+      "scala",
+      "java.lang",
+      "scala.Predef",
+      "scala.collection.immutable"
```

Comment:
**question:** should this be here? I don't think scala imports this by default, correct me if I'm wrong but some things like `List`/`Set`/`Map` are available because of aliases in `scala` / `scala.Predef`.

Reply:
You're probably right, I grabbed that code from the 0.18 branch.

If I remove it, some files in `bootstrapped` will get an additional import for things you mentionned: like `import scala.collection.immutable.Map`. So although it might be incorrect, the incorrect behaviour does not show because we don't use shapes that would be in `scala.collection.immutable` but not imported by the predef.

Reply:
Can we change occurrences of `NameRef("scala.collection.immutable", "Map")` to `NameRef("Predef", "Map")`? wouldn't that give us the right behavior too?

Reply:
btw. this sent me on a spiral of madness because I remembered some 2.12-specific issue with immutable/general collections. Thought it was about Set/Map, but it was about `Seq`:

from [2.13 release notes](https://github.com/scala/scala/releases/tag/v2.13.0):

```
Immutable scala.Seq
    Seq is now an alias for collection.immutable.Seq
        Before, it was an alias for the possibly-mutable collection.Seq.
```